src/hotspot/share/runtime/javaThread.cpp

```diff
@@ -414,6 +414,8 @@ JavaThread::JavaThread() :
   _vm_result(nullptr),
   _vm_result_2(nullptr),
 
+  _backoff_secondary_super_miss(0),
```

Comment:
Why don't you set it to `SecondarySuperMissBackoff` instead?

With the proposed shape of `MacroAssembler::check_klass_subtype_slow_path()` (`if ((x = (x-1)) >= 0) {... slow path ... }`, an overflow happens on the first access. It has an unfortunate consequence that '== 0' doesn't work anymore and, moreover, signed comparison is needed.

Reply:
Yeah, I guess there is no point in not initializing to `SSMB` right away. I was thinking that the first update should be accepted, as slowpath is probably correct in the optimistic case, and only the bad cases should pay. But given how this counter is shared by all callsites, that point is likely moot. Changed in new commit.